raiden/tests/unit/transfer/test_utils.py

```diff
@@ -17,3 +27,73 @@
 )
 def test_hash_balance_data(values, expected):
     assert hash_balance_data(values[0], values[1], values[2]) == expected
+
+
+def test_events_for_onchain_secretreveal_with_unfit_channels():
+    settle = factories.TransactionExecutionStatusProperties()
+    settled = factories.create(factories.NettingChannelStateProperties(settle_transaction=settle))
+    secret = factories.UNIT_SECRET
+    block_hash = factories.make_block_hash()
+
+    events = events_for_onchain_secretreveal(settled, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for settled channel"
+
+    settle = factories.replace(settle, result=TransactionExecutionStatus.FAILURE)
+    unusable = factories.create(factories.NettingChannelStateProperties(settle_transaction=settle))
+
+    events = events_for_onchain_secretreveal(unusable, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for unusable channel."
```

Comment:
Humm, what is an unusable channel here? This doesn't seem right to me.

Reply:
A channel with status `CHANNEL_STATE_UNUSABLE`, i. e. one with a failed settle or close transaction.

Reply:
Ohhh, it took me a while. Yeah, if the channel is settled it doesn't make sense to register the secret, the window is over. (For some reason I thought it was just a closed channel)